contracts/SprtNFTStorefront.cdc

```diff
@@ -496,9 +496,34 @@ pub contract SprtNFTStorefront {
 
             let listing <- self.listings.remove(key: listingResourceID)!
             assert(listing.getDetails().purchased == true, message: "listing is not purchased, only admin can remove")
+
+            for key in self.listings.keys {
+                let otherListing = &self.listings[key] as! &Listing{ListingPublic}
+
+                if self.cmpNFT(listing: &listing as &Listing{ListingPublic}, listing2: otherListing) {
+                    let listing <- self.listings.remove(key: key)
+                    destroy listing
+                }
+            }
+
             destroy listing
         }
 
+        pub fun cmpNFT(listing: &Listing{ListingPublic}, listing2: &Listing{ListingPublic}): Bool {
+            let listingDetails = listing.getDetails()
+            let listingDetails2 = listing2.getDetails()
+
+            if listingDetails.nftType != listingDetails2.nftType {
+                return false
+            }
+
+            if listingDetails.nftID != listingDetails2.nftID {
+                return false
+            }
+
+            return true
```

Comment:
cadence 에서 이게 가능한가요?

`
return listingDetails.nftType == listingDetails2.nftType and listingDetails.nftID == listingDetails2.nftID
`

Reply:
&&로 가능한데 정해진 formatting이 없고, 추후에 비교하는게 더 늘어났을때 관리가 힘들어질꺼 같아서 따로 뺐습니다

Reply:
This is possible, but the lack of Cadence's formatting features makes the code rather inconvenient to humans.
I thought it would be good to add it in an easy-to-understand way even if the conditions increase in the future.